### PR TITLE
Fail when tests executed without cli

### LINKED ISSUE #179: Better error when launched without AVA cli
Perhaps throw a more explicit error because at the moment it appears to crash and complain about lack of `process.send`:

```
> node test/index.js
Warning: a promise was rejected with a non-error: [object Undefined]
    at /Users/timoxley/Projects/tests/node_modules/ava/lib/runner.js:163:20
From previous event:
    at Runner.run (/Users/timoxley/Projects/tests/node_modules/ava/lib/runner.js:161:4)
    at Immediate._onImmediate (/Users/timoxley/Projects/tests/node_modules/ava/index.js:53:9)
    at processImmediate [as _immediateCallback] (timers.js:374:17)

Unhandled rejection TypeError: process.send is not a function
    at exit (/Users/timoxley/Projects/tests/node_modules/ava/index.js:42:10)
From previous event:
    at Immediate._onImmediate (/Users/timoxley/Projects/tests/node_modules/ava/index.js:53:15)
    at processImmediate [as _immediateCallback] (timers.js:374:17)
```

Related #97

### PR DESCRIPTION
**Pull request by vdemedes**
This PR fixes #179.

Here's the output:

screen shot 2015-11-11 at 7 45 53 pm

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
 var setImmediate = require('set-immediate-shim');
 var hasFlag = require('has-flag');
+var relative = require('path').relative;
 var serializeError = require('serialize-error');
 var Runner = require('./lib/runner');
+var log = require('./lib/logger');
 var runner = new Runner();
 
+// check if the test is being run without AVA cli
+var isForked = typeof process.send === 'function';
```
Comment (sindresorhus) [inline index.js]: Probably clearer to just check if the file was executed directly: `require.main === module`

File: index.js [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+
+if (!isForked) {
+	var path = relative(process.cwd(), process.argv[1]);
```
Comment (sindresorhus) [inline index.js]: `var path = relative('.', process.argv[1]);`

File: index.js [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+
+	log.write();
+	log.error('Test files must be executed using AVA CLI:\n\n    $ ava ' + path + '\n');
```
Comment (sindresorhus) [inline index.js]: using **the** AVA CLI

Comment (sindresorhus) [inline index.js]: > Test files must be run with the AVA CLI:

File: index.js (hunk 1 above)
Comment (vdemedes) [inline index.js]: But `index.js` is never required directly.

Comment (vdemedes) [inline index.js]: No matter if I use `node test.js` or `ava test.js`, `require.main === module` is always false.

Comment (sindresorhus) [inline index.js]: Oh, nvm. I'm sleepy...

File: index.js [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
 var runner = new Runner();
 
+// check if the test is being run without AVA cli
+var isForked = typeof process.send === 'function';
+
+if (!isForked) {
+	var path = relative('.', process.argv[1]);
+
+	log.write();
+	log.error('Test files must be run with the AVA CLI::\n\n    $ ava ' + path + '\n');
```
Comment (sindresorhus) [inline index.js]: double `::`

Comment (vdemedes) [inline index.js]: damn it

Comment (sindresorhus) [inline index.js]: and maybe add `chalk.dim('$')` and chalk.cyan to the command?

File: index.js [hunk 5]
```diff
@@ -1,10 +1,24 @@
...
 var runner = new Runner();
 
+// check if the test is being run without AVA cli
+var isForked = typeof process.send === 'function';
+
+if (!isForked) {
+	var path = relative('.', process.argv[1]);
+
+	log.write();
+	log.error('Test files must be run with the AVA CLI:\n\n    $ ava ' + path + '\n');
```
Comment (sindresorhus) [inline index.js]: and maybe add `chalk.dim('$')` and `chalk.cyan` to the command? just to be fancy.

File: index.js (hunk 4 above)
Comment (vdemedes) [inline index.js]: `chalk.dim('$') + ' ' + chalk.cyan('ava') + ' ' + path`?

File: index.js (hunk 5 above)
Comment (vdemedes) [inline index.js]: `chalk.dim('$') + ' ' + chalk.cyan('ava') + ' ' + path`?

Comment (vdemedes) [inline index.js]: screen shot 2015-11-11 at 8 02 24 pm

Comment (sindresorhus) [inline index.js]: I was thinking the full command:

```
chalk.dim('$') + ' ' + chalk.cyan('ava' + ' ' + path)
```

Comment (vdemedes) [inline index.js]: screen shot 2015-11-11 at 8 03 59 pm

Comment (sindresorhus) [inline index.js]: Perfect. Just `ava` in cyan looks better, but this is clearer.

**Comment by sindresorhus** [comment]
heart-universe

@vdemedes being productive today :D

**Comment by vdemedes** [comment]
giphy 1

**Pull request merged**